### ISSUE TITLE
Gradient reversal layer without custom operator

### ISSUE THREAD

**Issue by wellner** [dmlc/mxnet#3118]
I'm looking to implement unsupervised transfer learning architectures based on "gradient reversal" http://arxiv.org/abs/1505.07818 .   The idea is outlined nicely in this graphic from the paper:

<img width="565" alt="screen shot 2016-08-23 at 9 54 08 am" src="https://cloud.githubusercontent.com/assets/2430406/17894533/a9407af6-6917-11e6-89bc-9a8f7b09a268.png">

Any ideas on how to implement this approach without defining a custom gradient reversal operator/layer?  I'd prefer to use the Scala-Package which doesn't have support for custom layers yet.

**Comment by piiswrong** [comment]
use negative learning rate for some layers?

**Comment by wellner** [comment]
Ah, I think I can see how to do that via a combination of shared weights and explicitly setting learning rates (negative).  Thanks!!

**Comment by ifeherva** [comment]
I implemented this as custom operator currently, however I think this would be of interest to others. Would it make sense to implement it in Cpp as a standard or contrib operator? If yes I can PR it.

**Comment by gspell** [comment]
Hi @ifeherva, thanks for the work you've done in making a gradient reversal layer! Do you know how I can use it while using MX-net as my backend for Keras?
